### PR TITLE
fix: add a benchmark tests option

### PR DESCRIPTION
**Pull request by noob20000405**
Fix: https://github.com/casbin/casbin-cpp/issues/155

### Description

I moved all intensive benchmark tests functions like ```*Medium()``` and  ```*Large()``` to some new files called ```*_inten.cpp```. And added an option for intensive tests in the CMake file.

To do intensive tests, we need to use command ```cmake .. -DINTENSIVE_BENCHMARK=ON``` to switch on it.

### CODE REVIEW

**Comment by noob20000405** [comment]
@sheny1xuan @EmperorYP7  I didn't use macro but I think it meets your suggestion, if not, please give me some instructions. Thank you in advance ;))

File: tests/benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 add_definitions(-DCASBIN_PROJECT_DIR=${CMAKE_SOURCE_DIR})
 
+option(INTENSIVE_BENCHMARK OFF)
```
Comment (sheny1xuan) [inline tests/benchmarks/CMakeLists.txt:17]: Good job. Maybe adding a commet like [that](https://github.com/casbin/casbin-cpp/blob/abaa2174f0a4c6df2a25c21c388c15939f472655/CMakeLists.txt#L54) is better.

**Comment by hsluoyz** [comment]
@noob20000405 please do not raise new PRs for the same issue. You can always re-use the existing PR for doing anything: reworking commits, appending commits, etc.

**Comment by hsluoyz** [comment]
@EmperorYP7 plz review

File: tests/benchmarks/CMakeLists.txt (hunk 1 above)
Comment (noob20000405) [reply tests/benchmarks/CMakeLists.txt]: I followed your suggestion, please review again. Thanks.

**Review by sheny1xuan — APPROVED** [review]
LGTM！

**Review by EmperorYP7 — APPROVED** [review]
LGTM!

**Pull request merged**